### PR TITLE
Add geospatial filtering to spark job

### PR DESCRIPTION
**Pull request by andyphelps**
* Add parameter to allow WKT bounding
* Add parameter to specify output file prefix
* Add geospatial filter
* Refactor main analysis function

### CODE REVIEW

File: spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Config.scala [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
 case class Config(inputPath: String,
                   outputDirectory: String,
+                  outputFilenamePrefix: String,
                   isLocal: Boolean,
-                  resolution: Double)
+                  resolution: Double,
+                  boundingPolygon: Polygon)
 
 object Config {
+  private final val WHOLE_WORLD_WKT: Polygon = Polygon.apply(
```
Comment (JoelClemence) [inline spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Config.scala]:
```suggestion
  private final val WHOLE_WORLD_POLYGON: Polygon = Polygon.apply(
```

File: spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Config.scala [hunk 2]
```diff
@@ -19,38 +34,67 @@ object Config {
...
+    Try(WKT.read(wktInput)).isSuccess && WKT
+      .read(wktInput)
+      .as[Polygon]
+      .isDefined
+  }
+
+  private def parseWkt(wktInput: String): Polygon = {
+    WKT.read(wktInput).as[Polygon] match {
+      case Some(polygon) => polygon
+      case _             => WHOLE_WORLD_WKT
```
Comment (JoelClemence) [inline spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Config.scala]:
```suggestion
      case _             => WHOLE_WORLD_POLYGON
```

File: spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Config.scala [hunk 3]
```diff
@@ -19,38 +34,67 @@ object Config {
...
-        outputDirectory = "",
-        isLocal = false,
-        resolution = 1
-      )
+      Config(inputPath = "",
+             outputDirectory = "",
+             outputFilenamePrefix = DEFAULT_FILENAME_PREFIX,
+             isLocal = false,
+             resolution = 1,
+             boundingPolygon = WHOLE_WORLD_WKT)
```
Comment (JoelClemence) [inline spark-job/src/main/scala/uk/gov/ukho/ais/rasters/Config.scala]:
```suggestion
             boundingPolygon = WHOLE_WORLD_POLYGON)
```

File: spark-job/src/test/scala/uk/gov/ukho/ais/rasters/ComponentTest.scala [hunk 4]
```diff
@@ -20,7 +23,22 @@ class ComponentTest {
...
 
   private final val TEST_RESOLUTION = 1
-  private final val TOTAL_CELL_COUNT = 64800
+  private final val TOTAL_CELL_COUNT_WHOLE_WORLD_AT_1K_INCL_BORDER = 65884
+  private final val TOTAL_CELL_COUNT_LITTORAL_AT_1K_INCL_BORDER = 594
+  private final val LITTORAL_POLYGON =
+    "POLYGON((-20 65, -20 45, 5 45, 5 65, -20 65))"
+  private final val SE_TRIANGLE_LITTORAL_POLYGON =
+    "POLYGON((-20 45, 5 45, 5 65, -20 45))"
+  private final val WHOLE_WORLD_WKT: Polygon = Polygon.apply(
```
Comment (JoelClemence) [inline spark-job/src/test/scala/uk/gov/ukho/ais/rasters/ComponentTest.scala]:
```suggestion
  private final val WHOLE_WORLD_POLYGON: Polygon = Polygon.apply(
```

File: spark-job/src/test/scala/uk/gov/ukho/ais/rasters/ComponentTest.scala [hunk 5]
```diff
@@ -30,6 +48,12 @@ class ComponentTest {
   def beforeEach(): Unit = {
     val path: Path = Files.createTempDirectory("aisrastertest")
     tempOutputDir = path.toFile
+    testConfig = Config("",
+                        tempOutputDir.getAbsolutePath,
+                        "test-prefix",
+                        true,
+                        TEST_RESOLUTION,
+                        WHOLE_WORLD_WKT)
```
Comment (JoelClemence) [inline spark-job/src/test/scala/uk/gov/ukho/ais/rasters/ComponentTest.scala]:
```suggestion
                        WHOLE_WORLD_POLYGON)
```

File: spark-job/src/test/scala/uk/gov/ukho/ais/rasters/ComponentTest.scala (hunk 4 above)
Comment (andyphelps) [reply spark-job/src/test/scala/uk/gov/ukho/ais/rasters/ComponentTest.scala]: Well spotted!

File: ais-batch-lambda/src/main/java/uk/gov/ukho/aisbatchlambda/JobsProvider.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        if (instance == null) {
+            instance = new JobsProvider();
+        }
+
+        return instance;
+    }
+
+    public List<Job> getJobs() {
+        return Arrays.asList(
+                new Job(DEGREES_AT_EQUATOR_FOR_1KM_RESOLUTION, "littoral-1k", LITTORAL_BOUNDING_WKT),
```
Comment (ciaranevans) [inline ais-batch-lambda/src/main/java/uk/gov/ukho/aisbatchlambda/JobsProvider.java]: Everything looks fine to me!

- Disclaimer, the team said that they managed to downsample 1km to 5km themselves, seemed to sound like they're actually happy with 1km res and cropping, so watch this space.

**Comment by JoelClemence** [comment]
Feature no longer needed.

Also, there were some issues with GeoTrellis running the Spark job in EMR.

**Pull request closed without merging**